overlord/snapstate/check_snap.go

```diff
@@ -478,6 +479,22 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, _ snap.C
 		return nil
 	}
 
+	// do basic constraints check on the gadget
+	model := deviceCtx.Model()
+	if typ == snap.TypeGadget {
+		validateGadgetModelConstraints := func(model *asserts.Model, snapf snap.Container) error {
```

Comment:
I'm not sure why the local function?

Reply:
Silly me, this is from a previous iteration. I fixed this now.